File: javascript/bob/bob.js
function isUpperCase(l) {
    return /[A-Z\xdc\xc4\xdc]/.test(l);
}

function isLetter(l) {
    return /[A-Za-z\xfc\xe4\xdc]/.test(l);
}

function Bob() {
    this.answers = {
        question: 'Sure.',
        yell: 'Whoa, chill out!',
        nada: 'Fine. Be that way!',
        else: 'Whatever.'
    };
};

Bob.prototype.hey = function (input) {
    var inputChars = input.split('');
    var inputLetters = inputChars.filter(isLetter);
    if (inputLetters.length && inputLetters.every(isUpperCase)) {
        return this.answers.yell;
    }
    if (inputChars.slice(-1)[0] === '?') {
        return this.answers.question;
    }
    if (input.trim() === '') {
        return this.answers.nada;
    }
    return this.answers.else;

};

module.exports = Bob;

File: python/bob/bob.py
answers = {
	'question': 'Sure.',
	'yell': 'Whoa, chill out!',
	'nada': 'Fine. Be that way!',
	'whatever': 'Whatever.'
}

def hey(input):
    trimmed = input.strip()
    if trimmed.isupper():
    	return answers['yell']
    elif trimmed.endswith('?'):
    	return answers['question']
    elif not trimmed:
    	return answers['nada']
    else:
    	return answers['whatever']
